drivers/adc/adc_mcux_lpadc.c

```diff
@@ -412,27 +411,10 @@ static int mcux_lpadc_init(const struct device *dev)
 	}
 #endif
 
-#if !defined(CONFIG_SOC_SERIES_IMX_RT11XX)
-#if	defined(CONFIG_SOC_SERIES_IMX_RT6XX)
-
-	SYSCTL0->PDRUNCFG0_CLR = SYSCTL0_PDRUNCFG0_ADC_PD_MASK;
-	SYSCTL0->PDRUNCFG0_CLR = SYSCTL0_PDRUNCFG0_ADC_LP_MASK;
-	RESET_PeripheralReset(kADC0_RST_SHIFT_RSTn);
-	CLOCK_AttachClk(kSFRO_to_ADC_CLK);
-	CLOCK_SetClkDiv(kCLOCK_DivAdcClk, config->clock_div);
-
-#elif	defined(CONFIG_SOC_SERIES_IMX_RT5XX)
-	SYSCTL0->PDRUNCFG0_CLR = SYSCTL0_PDRUNCFG0_ADC_PD_MASK;
-	SYSCTL0->PDRUNCFG0_CLR = SYSCTL0_PDRUNCFG0_ADC_LP_MASK;
-	RESET_PeripheralReset(kADC0_RST_SHIFT_RSTn);
-	CLOCK_AttachClk(kFRO_DIV4_to_ADC_CLK);
-	CLOCK_SetClkDiv(kCLOCK_DivAdcClk, 1);
-
-#elif defined(CONFIG_SOC_LPC55S36)
-	CLOCK_SetClkDiv(kCLOCK_DivAdc0Clk, 2U, true);
-	CLOCK_AttachClk(kFRO_HF_to_ADC0);
-
-	/* Disable VREF power down */
+
+#if defined(CONFIG_SOC_LPC55S36)
```

Comment:
If we're going to remove the remaining SOC specific code, why include this? Wouldn't it be better to move this to the LPC55S36 SOC init?

Reply:
yes